File: BOJ/boj11723.cpp
/*
 * 11723 집합
 */
#include <bits/stdc++.h>
using namespace std;

int n, val, num;
string command;
void solve() {
	cin >> n;
	val = 0;
	while (n--) {
		cin >> command;
		if(command == "add") {
			cin >> num;
			if((val & (1 << num)) > 0) {//있으면 무시
				continue;
			} else {//없으면 추가
				val |= (1 << num);
			}
		} else if(command == "remove") {
			cin >> num;
			if((val & (1 << num)) > 0) {//있으면 삭제
				val ^= (1 << num);
			}
		} else if(command == "check") {
			cin >> num;
			if((val & (1 << num)) > 0) {
				cout << "1\n";
			} else {
				cout << "0\n";
			}
		} else if(command == "toggle") {
			cin >> num;
			if((val & (1 << num)) > 0) {//있으면 삭제
				val ^= (1 << num);
			} else {//없으면 추가
				val |= (1 << num);
			}
		} else if(command == "all") {
			for (int i = 1; i <= 20; i++) {
				val |= (1 << i);
			}	
		} else if(command == "empty") {
			val = 0;
		}
	}
}
int main(void){
	ios::sync_with_stdio(false);
    cin.tie(nullptr);
	solve();
	return 0;
}

File: BOJ/boj11723.java
/*
 * 11723 집합
 */
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class boj11723 {
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();
		//input
		int n = Integer.parseInt(br.readLine());
		int val = 0;
		for (int i = 0; i < n; i++) {
			String[] cmd = br.readLine().split(" ");
			if(cmd[0].equals("add")) {
				if((val & (1<<Integer.parseInt(cmd[1]))) > 0) {
					continue;
				} else {
					val |= (1<<Integer.parseInt(cmd[1]));
				}
			} else if(cmd[0].equals("remove")) {
				if((val & (1<<Integer.parseInt(cmd[1]))) > 0) {
					val ^= (1<<Integer.parseInt(cmd[1]));
				}
			} else if(cmd[0].equals("check")) {
				if((val & (1<<Integer.parseInt(cmd[1]))) > 0) {
					sb.append("1\n");
				} else {
					sb.append("0\n");
				}
			} else if(cmd[0].equals("toggle")) {
				if((val & (1<<Integer.parseInt(cmd[1]))) > 0) {
					val ^= (1<<Integer.parseInt(cmd[1]));
				} else {
					val |= (1<<Integer.parseInt(cmd[1]));
				}
			} else if(cmd[0].equals("all")) {
				for (int j = 1; j <= 20; j++) {
					if((val & (1<<j)) == 0) {
						val |= (1<<j);
					}
				}
			} else if(cmd[0].equals("empty")) {
				val = 0;
			}
		}
		System.out.println(sb);
		br.close();
	}
}